src/users/users.service.ts

```diff
@@ -43,6 +44,29 @@ export class UsersService {
     }
   }
 
+  async delete(userId: string) {
+    try {
+      const user = await this.usersRepository.findOne(userId);
+      if (user.role !== Roles.ADMIN) {
+        await this.usersRepository.delete(userId);
+        return true;
+      }
+      const admins = await this.usersRepository.find({
+        where: { role: 'admin' },
+      });
+      if (admins.length >= 2) {
+        await this.usersRepository.delete(userId);
+        return true;
+      }
+      throw new HttpException(
+        'You cannot delete the last admin in the database',
+        HttpStatus.BAD_REQUEST,
+      );
+    } catch (err) {
+      throw new HttpException(err, HttpStatus.BAD_REQUEST);
+    }
+  }
```

Comment:
I think you need to add a check that the user is not null.

Also, consider refactoring this method to use delete operation only once (seems like duplication). You can check the user role IS an admin and if so get the count of admins and after that, if it's equal to 1, throw an error. Otherwise, you will delete the user.

Reply:
I've fixed that part, please recheck it